### ISSUE TITLE
ServerSetupTest.setPortOffset has no impact on the port configuration 

### ISSUE THREAD

**Issue by sparsick** [greenmail-mail-test/greenmail#267]
Calling ServerSetupTest.setPortOffset has no impact on the port configuration. I wrote a test for demonstration.

Test code:

```java
ServerSetupTest.setPortOffset(1000);
ServerSetup smtp = ServerSetupTest.SMTP;
int port = smtp.getPort();
assertThat(port).isEqualTo(1025);
```
Failure message is

```
org.junit.ComparisonFailure: expected:<[1]025> but was:<[3]025>
Expected :1025
Actual   :3025
```
I think the reason is that `ServerSetupTest.SMTP` is initialized in a static constructor and a static constructor is always called before the first static method call.

**Versions I used:**
- Java 10
- Greenmail 1.5.9

**Comment by marcelmay** [comment]
Yes, the code with the static method ```setPortOffset``` is obviously broken.
The default test setups should be immutable, to avoid side effects between tests .

As a workaround, try to directly create a setup configuration:
```java
ServerSetup smtp = new ServerSetup(1025, "localhost", ServerSetup.PROTOCOL_SMTP);
```

Thx alot for reporting,
Marcel
